Tasks/Common/webdeployment-common/kuduutility.ts

```diff
@@ -169,3 +171,119 @@ async function createPhysicalPath(publishingProfile, physicalPath: string) {
+    headers['Content-Length'] = 0;
+
+    tl.debug('Removing file: ' + fileName + ' using publishUrl: ' + kuduDeploymentURL);
+    httpObj.get('DELETE', kuduDeploymentURL, headers, (error, response, contents) => {
+            if(error) {
+            defer.reject(tl.loc('FailedtoDeleteFileFromKuduError', fileName, kuduDeploymentURL));
+        }
+        else if(response.statusCode === 200 || response.statusCode === 204) {
+            tl.debug('file: ' + fileName + ' removed from path: ' + physicalPath);
+            defer.resolve('file removed from kudu');
+        }
+        else {
+            defer.reject(tl.loc('FailedtoDeleteFileFromKuduError', fileName, kuduDeploymentURL, response.statusCode, response.statusMessage));
+        }
+    });
+    return defer.promise;
+}
+
+async function runCommandOnKudu(publishUrl: string, physicalPath: string, headers, command: string) {
+    var defer = Q.defer<string>();
+    var kuduDeploymentURL = "https://" + publishUrl + "/api/command";
+    headers['Content-Type'] = 'application/json';
+    var jsonData = {
+        'command': command,
+        'dir': physicalPath
+    };
+
+    tl.debug('Executing Script on Kudu: ' + kuduDeploymentURL + '. Command: ' + command);
+    httpObj.send('POST', kuduDeploymentURL, JSON.stringify(jsonData), headers, (error, response, body) => {
+        if(error) {
+            defer.reject(tl.loc('FailedToRunScriptOnKuduError', kuduDeploymentURL, error));
+        }
+        else if(response.statusCode === 200) {
+            var responseBody = JSON.parse(body);
+            console.log(responseBody.Output);
+            if(responseBody.ExitCode === 0) {
+            console.log(responseBody.Error);
+                defer.resolve(tl.loc('SciptExecutionOnKuduSuccess'));
+            }
+            else {
+                tl.error(responseBody.Error);
+                defer.reject(tl.loc('SciptExecutionOnKuduFailed', responseBody.ExitCode, responseBody.Error));
+            }
+        }
+        else {
+            defer.reject(tl.loc('FailedToRunScriptOnKudu', kuduDeploymentURL, response.statusCode, response.statusMessage));
+        }
+    });
+    return defer.promise;
+}
+
+export async function runPostDeploymentScript(publishingProfile, scriptType, inlineScript, scriptPath, appOfflineFlag) {
+
+    var basicAuthToken = 'Basic ' + new Buffer(publishingProfile.userName + ':' + publishingProfile.userPWD).toString('base64');
+    var headers = {
+        'Authorization': basicAuthToken,
+        'If-Match': '*'
+    };
+    var scriptContent = '';
+
+    if(scriptType === 'Inline Script') {
+        scriptContent = inlineScript;
+    }
+    else {
+        var fileBuffer: Buffer = fs.readFileSync(scriptPath);
+        var encodingType = fileEncoding.detectFileEncoding( scriptPath, fileBuffer);
+        var scriptFileContent: string = fileBuffer.toString(encodingType[0]);
+        if(encodingType[1]) {
```

Comment:
Can we do the same that PowerShell inline has followed. I agree that we should have a max limit.